loader/loader.c

```diff
@@ -7120,13 +7120,46 @@ VKAPI_ATTR VkResult VKAPI_CALL terminator_EnumerateDeviceExtensionProperties(VkP
             *pPropertyCount = all_exts.count;
         }
     } else {
-        // Just return the count; need to add in the count of implicit layer extensions
-        // don't worry about duplicates being added in the count
-        *pPropertyCount = icd_ext_count;
+        // Have to find the number of unique extensions, ie no duplicates, as the properties list returned contains no duplicates.
 
+        // Find the current number of extensions (with duplicates). This is the upper bound for the ext_name_list
+        uint32_t max_exts_num = icd_ext_count;
         for (uint32_t i = 0; i < implicit_layer_list.count; i++) {
-            *pPropertyCount += implicit_layer_list.list[i].device_extension_list.count;
+            max_exts_num += implicit_layer_list.list[i].device_extension_list.count;
         }
+
+        const struct loader_instance *inst = icd_term->this_instance;
+
+        uint32_t ext_name_count = 0;
+        char **ext_name_list = loader_instance_heap_alloc(inst, sizeof(char *) * max_exts_num, VK_SYSTEM_ALLOCATION_SCOPE_INSTANCE);
```

Comment:
Not that it actually makes a difference to the functionality, but `VK_SYSTEM_ALLOCATION_SCOPE_INSTANCE` is wrong here because you're only only keeping the memory around for the duration of the command. It would be more correct to use `VK_SYSTEM_ALLOCATION_SCOPE_COMMAND`.

Reply:
Fixed